src/Init/Data/Array/Lemmas.lean

```diff
@@ -220,7 +220,7 @@ theorem getElem?_len_le (a : Array α) {i : Nat} (h : a.size ≤ i) : a[i]? = no
 theorem getD_get? (a : Array α) (i : Nat) (d : α) :
   Option.getD a[i]? d = if p : i < a.size then a[i]'p else d := by
   if h : i < a.size then
-    simp [setD, h, getElem?]
+    simp [setD, h, getElem?, decidableGetElem?]
```

Comment:
I haven't tried it, but I think better than unfolding would be
```suggestion
    simp [setD, h, getElem?_def]
```
because that is why we have the `LawfulGetElem` class.